cardano-api/src/Cardano/Api/Fees.hs

```diff
@@ -837,24 +859,45 @@ makeTransactionBodyAutoBalance eraInMode systemstart history pparams
    era' :: CardanoEra era
    era' = cardanoEra
 
-   balanceCheck :: Lovelace -> Lovelace -> Either TxBodyErrorAutoBalance ()
-   balanceCheck minUTxOValue balance
+   balanceCheck :: Lovelace -> Either TxBodyErrorAutoBalance ()
+   balanceCheck balance
```

Comment:
This no longer checks that the balance is over the min utxo value. The check earlier
```
    mapM_ (`checkMinUTxOValue` pparams) $ txOuts txbodycontent1
```
does *not* cover the case of the balance, because the change output with the balance is not included in `txbodycontent1`.

I would keep the original check
```
    | balance < minUTxOValue = Left (TxBodyErrorAdaBalanceTooSmall balance)
```

I would also suggest splitting the error case into two because they are different errors with different causes, different solutions and should have different error messages. The `TxBodyErrorAdaBalanceTooSmall` should be for the case of the balance being too small.

We should have a separate error for the other user-specified outputs being too small, i.e. used by `checkMinUTxOValue` in
```
    mapM_ (`checkMinUTxOValue` pparams) $ txOuts txbodycontent1
```